monitoring/prometheus/prometheus.go

```diff
@@ -0,0 +1,195 @@
+		}, []string{"type", "service", "method"})
+
+	serverHandledHistogram = prom.NewHistogramVec(
+		prom.HistogramOpts{
+			Namespace: "grpc",
+			Subsystem: "server",
+			Name:      "rpc_handled",
+			Help:      "Histogram of response latency of RPC that had been application-level handled by the server.",
+			Buckets:   prom.DefBuckets,
+		}, []string{"type", "service", "method", "code"})
+
+	serverErred = prom.NewCounterVec(
+		prom.CounterOpts{
+			Namespace: "grpc",
+			Subsystem: "server",
+			Name:      "rpc_erred_total",
+			Help:      "Total number of RPC that had failed on the RPC layer on the server.",
+		}, []string{"type", "service", "method", "error"})
+
+
+	clientStartedCounter = prom.NewCounterVec(
+		prom.CounterOpts{
+			Namespace: "grpc",
+			Subsystem: "client",
+			Name:      "rpc_started_total",
+			Help:      "Total number of RPCs started by the client.",
+		}, []string{"type", "service", "method"})
+
+	clientStreamMsgReceived = prom.NewCounterVec(
+		prom.CounterOpts{
+			Namespace: "grpc",
+			Subsystem: "client",
+			Name:      "rpc_msg_received_total",
+			Help:      "Total number of RPC stream messages received by the client.",
+		}, []string{"type", "service", "method"})
+
+	clientStreamMsgSent = prom.NewCounterVec(
+		prom.CounterOpts{
+			Namespace: "grpc",
+			Subsystem: "client",
+			Name:      "rpc_msg_sent_total",
+			Help:      "Total number of RPC stream messages sent by the client.",
+		}, []string{"type", "service", "method"})
+
+	clientHandledHistogram = prom.NewHistogramVec(
+		prom.HistogramOpts{
+			Namespace: "grpc",
+			Subsystem: "client",
+			Name:      "rpc_handled",
+			Help:      "Histogram of of RPC latency on the client side",
+			Buckets:   prom.DefBuckets,
+		}, []string{"type", "service", "method", "code"})
+
+	clientErred = prom.NewCounterVec(
+		prom.CounterOpts{
+			Namespace: "grpc",
+			Subsystem: "client",
+			Name:      "rpc_erred_total",
+			Help:      "Total number of RPC that had failed on the RPC layer on the client.",
+		}, []string{"type", "service", "method", "error"})	
+)
+
+func init() {
+	prom.MustRegister(serverStartedCounter)
+	prom.MustRegister(serverStreamMsgReceived)
+	prom.MustRegister(serverStreamMsgSent)
+	prom.MustRegister(serverHandledHistogram)
+	prom.MustRegister(serverErred)
+	prom.MustRegister(clientStartedCounter)
+	prom.MustRegister(clientStreamMsgReceived)
+	prom.MustRegister(clientStreamMsgSent)
+	prom.MustRegister(clientHandledHistogram)
+	prom.MustRegister(clientErred)
+}
+
+type ServerMonitor struct {
+}
+
+func (m *ServerMonitor) NewServerMonitor(rpcType monitoring.RpcType, fullMethod string) monitoring.RpcMonitor {
+	r := &serverRpcMonitor{rpcType: rpcType, startTime: time.Now()}
+	r.serviceName, r.methodName = splitMethodName(fullMethod)
+	serverStartedCounter.WithLabelValues(string(r.rpcType), r.serviceName, r.methodName).Inc()
+	return r
+}
+
+type serverRpcMonitor struct {
+	rpcType     monitoring.RpcType
```

Comment:
Ah, you use the type here. disregard my comment above about exporting this type.